gateway_test.go

```diff
@@ -24,9 +24,66 @@ func init() {
 	runningTests = true
 }
 
+// to register to, but never used
 var discardMuxer = mux.NewRouter()
 
+const (
+	// we need a static port so that the urls can be used in static
+	// test data and the requests aren't randomized for checksums
+	// port 16500 should be obscure and unused
+	testHttpListen = "127.0.0.1:16500"
```

Comment:
If you get rid of fixed port, you can use ":0", so OS will assign random free port, and getting it back using `server.Addr`. The same way you can get rid of global http server.

It can be a separate PR later if you prefer. Or maybe we can leave it as it is.